### DISCUSSION TITLE
Does Ginkgo run on ARM64?

### DISCUSSION THREAD

**Discussion by NikMeier** [ginkgo-project/ginkgo#1560] [Q&A]
When attempting to compile examples/heat-equation I get errors such as:
 - Undefined symbols for architecture arm64:
or
 - ld: symbol(s) not found for architecture arm64
Does ginkgo run on the arm64 architecture (on which I am attempting to compile)?

**Comment by upsj** [comment]
I can successfully build Ginkgo with gcc or AppleClang on Apple M2 CPUs, so it should work. We'd need some more details on your environment and the exact error message to figure out what is going on. Can you post the full output of a fresh `cmake` invocation and the output of the build step?

**Comment by tcojean** [comment]
It's late, but I want to add that our MacOS CI runs on ARM64 with AppleClang as well, so there should be ways to make it work.

**Comment by tpadioleau** [comment]
Hi, I could also get some linker errors on arm64 in version 1.8, see my library CI: https://github.com/CExA-project/ddc/actions/runs/16135699076/job/45531435306.

It seems to pass in version 1.9.

**Reply by yhmtsai** [reply]
Hi, could you try adding `-undefined dynamic_lookup` in MacOS?
It might also be related to https://github.com/ginkgo-project/ginkgo/pull/1724

**Reply by tpadioleau** [reply]
Thanks for the quick feedback, it seems to work https://github.com/CExA-project/ddc/actions/runs/16139324030/job/45542944999.

I have added `CMAKE_SHARED_LINKER_FLAGS="-undefined dynamic_lookup"`.

**Reply by yhmtsai** [reply]
Thanks for confirmation. 
I also do it via `CMAKE_CXX_FLAGS` with `-Xlinker -undefined -Xlinker dynamic_lookup` although it will give tons of warning about unused argument during compilation.
I hope the workaround is enough for your project when using Ginkgo 1.8.0.
After Ginkgo 1.9.0, you do not need to add this flag anymore

**Comment by yhmtsai** [comment]
Hi @NikMeier , could you check again the issue still happens after Ginkgo 1.9.0 on your side?
From the discussion with @tpadioleau , https://github.com/ginkgo-project/ginkgo/pull/1724 should fix the issue, so Ginkgo 1.9.0 should not have that issue.
If you still need to use Ginkgo 1.8.0, please try the workaround in https://github.com/ginkgo-project/ginkgo/discussions/1560#discussioncomment-13692311
However, from your first post about heat-equation, it might be related to OpenCV installation from your system.
if you still have the issue, please also attach the compilation error log, which might help us to figure out the reason.
